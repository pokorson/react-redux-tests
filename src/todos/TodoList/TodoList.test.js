import React from 'react';
import { shallow } from 'enzyme';
import uuidv1 from 'uuid/v1';
import toJson from 'enzyme-to-json';

describe('TodoList component', () => {
  jest.mock('../api/api');
  const TodoList = require('./TodoList').default;
  const TodoListItem = require('../TodoListItem').default;

  const todos = [
    { id: 1, title: 'test1', completed: false },
    { id: 2, title: 'test2', completed: true },
    { id: 3, title: 'test3', completed: false },
  ];

  it('render TodoListItem list', () => {
    const component = shallow(<TodoList />);
    component.setState({ todos: todos });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders only completed todos when onlyCompleted is true', () => {
    const component = shallow(<TodoList />);
    component.setState({ todos: todos, onlyCompleted: true });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('toggles onlyCompleted flag on checkbox click', () => {
    const component = shallow(<TodoList />);
    component.find('input').simulate('change');
    expect(component.state()).toHaveProperty('onlyCompleted', true);
    component.find('input').simulate('change');
    expect(component.state()).toHaveProperty('onlyCompleted', false);
  });

  it('calls completeTodo action from API and update completed in todo', done => {
    const component = shallow(<TodoList />);
    component.setState({ todos });
    component
      .instance()
      .completeTodo(todos[0].id)
      .then(() => {
        expect(component.state().todos[0].completed).toEqual(!todos[0].completed);
        done();
      });
  });

  it('calls removeTodo action from API and updates todos', done => {
    const component = shallow(<TodoList />);
    component.setState({ todos });
    component
      .instance()
      .removeTodo(todos[0].id)
      .then(() => {
        expect(component.find(TodoListItem)).toHaveLength(2);
        expect(component.state().todos).not.toContain(todos[0]);
        done();
      });
  });

  it('calls addTodo action from API and updates todos', done => {
    const component = shallow(<TodoList />);
    component.setState({ todos });
    const newTodo = { id: 4, title: 'new test', completed: false };
    component
      .instance()
      .addTodo(newTodo)
      .then(() => {
        expect(component.state().todos).toHaveLength(4);
        expect(component.state().todos).toContain(newTodo);
        done();
      });
  });
});
