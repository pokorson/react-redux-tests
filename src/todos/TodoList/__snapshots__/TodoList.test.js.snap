// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component render TodoListItem list 1`] = `
<div>
  <TodoForm
    onSubmit={[Function]}
  />
  <div>
    <label>
      only completed
    </label>
    <input
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
  </div>
  <div
    className="todos-list"
  >
    <TodoListItem
      completed={false}
      id={1}
      onComplete={[Function]}
      onRemove={[Function]}
      title="test1"
    />
    <TodoListItem
      completed={true}
      id={2}
      onComplete={[Function]}
      onRemove={[Function]}
      title="test2"
    />
    <TodoListItem
      completed={false}
      id={3}
      onComplete={[Function]}
      onRemove={[Function]}
      title="test3"
    />
  </div>
</div>
`;

exports[`TodoList component renders only completed todos when onlyCompleted is true 1`] = `
<div>
  <TodoForm
    onSubmit={[Function]}
  />
  <div>
    <label>
      only completed
    </label>
    <input
      onChange={[Function]}
      type="checkbox"
      value={true}
    />
  </div>
  <div
    className="todos-list"
  >
    <TodoListItem
      completed={true}
      id={2}
      onComplete={[Function]}
      onRemove={[Function]}
      title="test2"
    />
  </div>
</div>
`;
